src/sentry/models/project.py

```diff
@@ -407,3 +407,46 @@ def get_security_token(self):
 
     def get_lock_key(self):
         return 'project_token:%s' % self.id
+
+    def copy_settings_from(self, project_id):
+        """
+        Copies project level settings of the inputted project
+        - General Settings
+        - ProjectTeams
+        - Alerts Settings and Rules
+        - EnvironmentProjects
+        - ProjectOwnership Rules and settings
+        - Project Inbound Data Filters
+
+        Returns True if the settings have successfully been copied over
+        Returns False otherwise
+        """
+        from sentry.models import (
+            EnvironmentProject, ProjectOption, ProjectOwnership, Rule
+        )
+        model_list = [EnvironmentProject, ProjectOwnership, ProjectTeam, Rule]
+
+        project = Project.objects.get(id=project_id)
+        try:
+            with transaction.atomic():
+                for model in model_list:
+                    # remove all previous project settings
+                    model.objects.filter(
+                        project_id=self.id,
+                    ).delete()
+
+                    # add settings from other project to self
+                    for setting in model.objects.filter(
+                        project_id=project_id
+                    ):
+                        setting.pk = None
+                        setting.project_id = self.id
+                        setting.save()
+
+                options = ProjectOption.objects.get_all_values(project=project)
+                for key, value in six.iteritems(options):
+                    self.update_option(key, value)
+
+        except IntegrityError:
+            return False
```

Comment:
Do you want any logging or metrics when this fails? Should we send a different response message to the user as their request may not have been completed in its entirety.

Reply:
@markstory correct me if I'm wrong, but if it's in a transaction, and something goes wrong, then all changes are reverted and nothing was changed. That's what I was going for here. Am I missing something?

Reply:
I think logging is a good idea. I have no idea what message to give the user though... outside of a 500 if this fails. :/

Reply:
Yes all these changes are in a transaction, but the project creation and project option setup happens outside this transaction block. The requested project will end up being created, but not with the copied settings.

In the case of an integrity error you could raise 400 or 409 instead of a 500.

Reply:
This is only used in the `put`, so I don't think we care too much about the project already existing. It could be weird if they pass other values to the endpoint other than to copy from settings though.